src/GraphQL.NewtonsoftJson/GraphQLRequestListJsonConverter.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Collections.Generic;
+using GraphQL.Transport;
+using Newtonsoft.Json;
+
+namespace GraphQL.NewtonsoftJson
+{
+    /// <summary>
+    /// A custom JsonConverter for reading or writing a <see cref="GraphQLRequest"/> object.
+    /// </summary>
+    public class GraphQLRequestListJsonConverter : JsonConverter
+    {
+        /// <inheritdoc/>
+        public override bool CanConvert(Type objectType)
+        {
+            return (
+                objectType == typeof(IEnumerable<GraphQLRequest>) ||
+                objectType == typeof(ICollection<GraphQLRequest>) ||
+                objectType == typeof(IReadOnlyCollection<GraphQLRequest>) ||
+                objectType == typeof(IReadOnlyList<GraphQLRequest>) ||
+                objectType == typeof(IList<GraphQLRequest>) ||
+                objectType == typeof(List<GraphQLRequest>) ||
+                objectType == typeof(GraphQLRequest[]));
+        }
+
+        /// <inheritdoc/>
+        public override bool CanRead => true;
+
+        /// <inheritdoc/>
+        public override bool CanWrite => true;
+
+        /// <inheritdoc/>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            var requests = (IEnumerable<GraphQLRequest>)value;
+            writer.WriteStartArray();
+            foreach (var request in requests)
+            {
+                serializer.Serialize(writer, request);
+            }
+            writer.WriteEndArray();
+        }
+
+        /// <inheritdoc/>
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
```

Comment:
Side note - what is existingValue ? Is it `null`? I think it may be not null for some merge cases.

Reply:
I found - https://stackoverflow.com/questions/24347613/what-is-the-existingvalue-parameter-used-for-in-the-readjson-method-of-a-jsoncon

Reply:
🤷‍♂️

https://stackoverflow.com/questions/24347613/what-is-the-existingvalue-parameter-used-for-in-the-readjson-method-of-a-jsoncon

Reply:
Doesn't seem to be anything we need worry about.